mifospay/src/main/java/org/mifos/mobilewallet/mifospay/settings/ui/SettingsScreen.kt

```diff
@@ -0,0 +1,142 @@
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(20.dp),
+                        style = TextStyle(fontSize = 17.sp, color = mifosText)
+                    )
+                }
+            }
+
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .weight(1f)
+            ) {}
+
+            Row(modifier = Modifier.padding(8.dp)) {
+                MifosItemCard(
+                    onClick = { currentDialog = DialogType.DISABLE_ACCOUNT },
+                    colors = CardDefaults.cardColors(Color.Black)
+                ) {
+                    Text(
+                        text = stringResource(id = R.string.disable_account).uppercase(),
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(20.dp),
+                        style = styleSettingsButton
+                    )
+                }
+            }
+
+            Row(modifier = Modifier.padding(8.dp)) {
+                MifosItemCard(
+                    onClick = { currentDialog = DialogType.LOGOUT },
+                    colors = CardDefaults.cardColors(Color.Black)
+                ) {
+                    Text(
+                        text = stringResource(id = R.string.log_out).uppercase(),
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(20.dp),
+                        style = styleSettingsButton
+                    )
+                }
+            }
+        }
+    }
+
+    if (currentDialog != DialogType.NONE) {
+        MifosDialogBox(
+            showDialog = true,
+            onDismiss = { currentDialog = DialogType.NONE },
+            title = when (currentDialog) {
+                DialogType.DISABLE_ACCOUNT -> R.string.alert_disable_account
+                DialogType.LOGOUT -> R.string.log_out_title
+                else -> R.string.empty
+            },
+            message = when (currentDialog) {
+                DialogType.DISABLE_ACCOUNT -> R.string.alert_disable_account_desc
+                else -> R.string.empty
+            },
+            confirmButtonText = when (currentDialog) {
+                DialogType.DISABLE_ACCOUNT -> R.string.ok
+                DialogType.LOGOUT -> R.string.yes
+                else -> R.string.empty
+            },
+            dismissButtonText = when (currentDialog) {
+                DialogType.DISABLE_ACCOUNT -> R.string.cancel
+                DialogType.LOGOUT -> R.string.no
+                else -> R.string.empty
+            },
+            onConfirm = {
+                when (currentDialog) {
+                    DialogType.DISABLE_ACCOUNT -> disable()
+                    DialogType.LOGOUT -> logout()
+                    else -> {}
+                }
+                currentDialog = DialogType.NONE
```

Comment:
I have converted the dialog box to compose @therajanmaurya . I am currently using enums to handle the disable button and logout button. However earlier I had used the below approach to open the dialog box
```
    var showDisableDialog by remember { mutableStateOf(false) }
    var showLogoutDialog by remember { mutableStateOf(false) }

    MifosDialogBox(
        showDialog = showDisableDialog,
        onDismiss = { showDisableDialog = false },
        title = R.string.alert_disable_account,
        message = R.string.alert_disable_account_desc,
        confirmButtonText = R.string.ok,
        dismissButtonText = R.string.cancel,
        onConfirm = { disable.invoke() }
    )

    MifosDialogBox(
        showDialog = showLogoutDialog,
        onDismiss = { showLogoutDialog = false },
        title = R.string.log_out_title,
        confirmButtonText = R.string.yes,
        dismissButtonText = R.string.no,
        onConfirm = { logout.invoke() }
    )
    
    Hence I wanted your opinion on this

Reply:
Approach is good but let's have separate method for logout and disable because it more code with condition rather than separate method.

Reply:
Ok, so I can use the earlier approach?